src/replace-libraries-ngpackagr-builder.ts

```diff
@@ -0,0 +1,81 @@
+import {AngularJson} from './types/angular-json';
+import {cloneDeep} from 'lodash-es';
+import {AngularVersion} from './types/angular-version';
+import {angularVersionComparer} from './angular-version-comparer';
+import {ProjectConfiguration} from './types/project-configuration';
+import {ArchitectConfiguration} from './types/architect-configuration';
+
+export const preAngular10_1NgPackagrBuilder =
+  '@angular-devkit/build-ng-packagr:build';
+export const angular10_1AndUpNgPackagrBuilder =
+  '@angular-devkit/build-angular:ng-packagr';
+
+const angular10_1: AngularVersion = '10.1.x';
+
+/**
+ * Replace all libraries builder using ng-packagr with the correct builder given the Angular version
+ *
+ */
+export function replaceLibrariesNgPackagrBuilder(
+  angularVersion: AngularVersion,
+  angularJson: AngularJson
+): AngularJson {
+  const modifiedAngularJson = cloneDeep(angularJson);
+
+  const projectsAndtargetsWithNgPackagrBuilder = getNameOfProjectsAndTargetsUsingNgpackagr(
+    angularJson
+  );
+
+  for (const [
+    projectName,
+    targetName
+  ] of projectsAndtargetsWithNgPackagrBuilder) {
+    modifiedAngularJson.projects[projectName].architect[
+      targetName
+    ].builder = getCorrectNgPackgrBuilder(angularVersion);
+  }
+
+  return modifiedAngularJson;
+}
+
+/**
+ * Return a list with all project/target name tuples using a ng-packagr builder.
+ *
+ */
+function getNameOfProjectsAndTargetsUsingNgpackagr(
+  workspace: AngularJson
+): [string, string][] {
+  return (
+    Object.entries(workspace.projects)
+      .filter(([, projectConfig]) => projectConfig.projectType === 'library')
+      .map(([projectName, library]) => {
+        return [
+          projectName,
+          findProjectTargetWithNgPackagrBuilder(library)
+        ] as [string, [string, ArchitectConfiguration]];
+      })
+      // eslint-disable-next-line eqeqeq
+      .filter(([, target]) => target != undefined)
```

Comment:
Question: Why are you disabling the rule rather than making a strict equals comparison?
```suggestion
      .filter(([, target]) => target !== undefined)
```